src/server/pfs/server/driver.go

```diff
@@ -458,8 +458,18 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
 		Description: description,
 	}
 
-	//  BuildCommit case: if the caller passed a tree reference with the commit
-	// contents then retrieve the full tree so we can compute its size
+	// FinishCommit case. We need to create AND finish 'newCommit' in two cases:
+	// 1. PutFile has been called on a finished commit (records != nil), and
+	//    we want to apply 'records' to the parentCommit's HashTree, use that new
+	//    HashTree for this commit's filesystem, and then finish this commit
+	// 2. BuildCommit has been called by migration (treeRef != nil) and we
+	//    want a new, finished commit with the given treeRef
+	// - In either case, store this commit's HashTree in 'tree', so we have its
+	//   size, and store a pointer to the tree (in object store) in 'treeRef', to
+	//   put in newCommitInfo.Tree.
+  // - We also don't want to resolve 'branch' or 'parent.ID' (if it's a branch)
+  //   outside the txn below, so the 'PutFile' case is handled (by computing
```

Comment:
nit: fix mixed tabbing

Reply:
Fixed; ran gofmt (which I just forgot to do before committing :sweat_smile:)